R/functions.R

```diff
@@ -395,224 +382,260 @@ vwparams <- function(vwmodel, name) {
-  } else {
-      algorithm_check_type <- get(paste0(params$algorithm, "_check"))
-      params$optimization_params <- check_param_values(
-          input = params$optimization_params,
-          check = c(algorithm_check_type, optimization_check)
-      )
-  }
-  
-  # Cache should be created, if passes > 1
-  if(params$general_params$passes > 1) {
-      params$general_params$cache <- TRUE
-  }
-  # Return validated parameters
-  return(list(algorithm = params$algorithm,
-              general_params = params$general_params,
-              optimization_params = params$optimization_params,
-              reductions = params$reductions))
-}
-
-.create_parameters_string <- function(params) {
-    params_to_strings <- function(i) {
-        if(is.na(flat_params[[i]]) | flat_params[[i]] == "") {
-            return("")
-        };
-        if(is.logical(flat_params[[i]][[1]]) & flat_params[[i]][[1]] == TRUE) {
-            return(paste0("--",i))
-        }; 
-        if(is.logical(flat_params[[i]][[1]]) & flat_params[[i]][[1]] == FALSE) {
-            return("")
-        } else {
-            return(paste0("--",i," ",flat_params[[i]]))
+    # Construct labels for multilabel examples: 1:0.3 2:0.3 3:0.3
+    constructLabels <- function(targets, num_row) {
+        # Initialize output vector
+        tmp <- rep("", num_row)
+        # Iterate cost vectors names
+        for(i in 1:length(targets)) {
+            elem_targets <- eval(get(targets[i], envir=parent.frame(n=1)))
+            # Iterate cost vectors individualy (nrow of data)
+            for (j in 1:num_row) {
+                ifelse(is.na(elem_targets[j]),
+                       "",
+                       tmp[j] <- paste(tmp[j],
+                                       paste(i, elem_targets[j], sep = ":")))
+            }
+        }
+        trimws(tmp)
+    }
+    # Construct labels for Context Bandit: 1:0.3:0.3 2:0.3:0.3 3:0.3:0.3
+    constructLabelsCB <- function(targets, probabilities, num_row) {
+        # Initialize output vector
+        tmp <- rep("", num_row)
+        # Iterate cost vectors names
+        for(i in 1:length(targets)) {
+            elem_targets <- eval(get(targets[i], envir=parent.frame(n=1)))
+            elem_probability <- eval(get(probabilities[i], envir=parent.frame(n=1)))
+            # Iterate cost vectors individualy (nrow of data)
+            for (j in 1:num_row) {
+                ifelse(is.na(elem_targets[j]),
+                       "",
+                       tmp[j] <- paste(tmp[j],
+                                       paste(i, elem_targets[j], elem_probability[j], sep = ":")))
+            }
         }
+        trimws(tmp)
     }
     
-    flatten <- function(x) {
-        repeat {
-            if(!any(vapply(x,is.list, logical(1)))) return(x)
-            x <- Reduce(c, x)
+    # namespace load with a yaml file
+    if (typeof(namespaces) == "character" && length(namespaces) == 1 &&
```

Comment:
This can be greatly simplified to:

```r
if(requireNamespace("yaml", quietly = TRUE)) {
    namespaces <- yaml::yaml.load_file(namespaces)
} else {
    stop("The 'yaml' package is needed.", .Call=FALSE)
}
```

Reply:
Yes, I agree, will replace this